osbenchmark/worker_coordinator/runner.py

```diff
@@ -677,29 +677,21 @@ class ForceMerge(Runner):
     """
 
     async def __call__(self, opensearch, params):
-        # pylint: disable=import-outside-toplevel
-        import opensearchpy
         max_num_segments = params.get("max-num-segments")
         mode = params.get("mode")
         merge_params = self._default_kw_params(params)
         if max_num_segments:
             merge_params["max_num_segments"] = max_num_segments
         if mode == "polling":
-            complete = False
-            try:
-                request_context_holder.on_client_request_start()
-                await opensearch.indices.forcemerge(**merge_params)
-                request_context_holder.on_client_request_end()
-                complete = True
-            except opensearchpy.ConnectionTimeout:
-                pass
-            while not complete:
-                await asyncio.sleep(params.get("poll-period"))
-                tasks = await opensearch.tasks.list(params={"actions": "indices:admin/forcemerge"})
-                if len(tasks["nodes"]) == 0:
-                    # empty nodes response indicates no tasks
+            merge_params["wait_for_completion"] = "false"
+            request_context_holder.on_client_request_start()
+            response_task = await opensearch.indices.forcemerge(**merge_params)
+            while True:
+                task = await opensearch.tasks.get(task_id=response_task['task'])
+                if task and task['completed']:
```

Comment:
Do not fall into an infinite loop if the `get` returns no task for some reason and provide an exit.

Reply:
Ack